go-client/api/swagger.yaml

```
@@ -0,0 +1,90 @@
+openapi: 3.0.2
+info:
+  title: Web Server
+  description: A simple web server for managing users
+  contact:
+    email: dmanor@redhat.com
+  version: "1.0"
+servers:
+- url: http://127.0.0.1:3000
+paths:
+  /api:
+    post:
+      summary: Sends the user's data to the server
+      operationId: addUser
+      parameters:
+      - name: name
+        in: query
+        description: user's name
+        required: true
+        style: form
+        explode: true
+        schema:
+          type: string
+      - name: email
+        in: query
+        description: user's email
+        required: true
+        style: form
+        explode: true
+        schema:
+          type: string
+      responses:
+        "200":
+          description: User added successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User/properties/userId'
+        "400":
+          description: A problem occured while trying to add a user
+  /api/{userId}:
+    delete:
+      summary: Deletes a user from DB
+      operationId: deleteUser
+      parameters:
+      - name: userId
```

just id, otherwise it feels a bit redundant 
```suggestion
      - name: id
```